modules/packages/LinearAlgebra.chpl

```diff
@@ -806,6 +806,70 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
   return A;
 }
 
+pragma "no doc"
+proc inv(ref A: [?Adom] ?eltType) throws
+where !usingLAPACK
+{
+  if Adom.rank != 2 then
+    halt("Wrong rank for matrix inverse");
```

Comment:
This can be a `compilerError`

Reply:
I believe `isSquare` covers this requirement as well, so we could drop this check.